gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,793 @@
+ * To get an {@code Operation} object with the most recent information, use
+ * {@link #reload(Compute.OperationOption...)}.
+ */
+public final class Operation implements Serializable {
+
+  private static final long serialVersionUID = -8979001444590023899L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private transient Compute compute;
+  private final ComputeOptions options;
+  private final String id;
+  private final OperationId operationId;
+  private final Long creationTimestamp;
+  private final String clientOperationId;
+  private final String operationType;
+  private final String targetLink;
+  private final String targetId;
+  private final Status status;
+  private final String statusMessage;
+  private final String user;
+  private final Integer progress;
+  private final Long insertTime;
+  private final Long startTime;
+  private final Long endTime;
+  private final List<OperationError> errors;
+  private final List<OperationWarning> warnings;
+  private final Integer httpErrorStatusCode;
+  private final String httpErrorMessage;
+  private final String description;
+
+  /**
+   * Status of an operation.
+   */
+  public enum Status {
+    PENDING,
+    RUNNING,
+    DONE
+  }
+
+  /**
+   * An error that can occur during the processing of a Google Compute Engine operation.
+   */
+  public static final class OperationError implements Serializable {
+
+    static final Function<com.google.api.services.compute.model.Operation.Error.Errors,
+        OperationError> FROM_PB_FUNCTION = new Function<
+            com.google.api.services.compute.model.Operation.Error.Errors, OperationError>() {
+          @Override
+          public OperationError apply(
+              com.google.api.services.compute.model.Operation.Error.Errors pb) {
+            return OperationError.fromPb(pb);
+          }
+        };
+    static final Function<OperationError,
+        com.google.api.services.compute.model.Operation.Error.Errors> TO_PB_FUNCTION =
+        new Function<OperationError,
+            com.google.api.services.compute.model.Operation.Error.Errors>() {
+          @Override
+          public com.google.api.services.compute.model.Operation.Error.Errors apply(
+              OperationError operation) {
+            return operation.toPb();
+          }
+        };
+
+    private static final long serialVersionUID = -1155314394806515873L;
+
+    private final String code;
+    private final String location;
+    private final String message;
+
+    OperationError(String code, String location, String message) {
+      this.code = code;
+      this.location = location;
+      this.message = message;
+    }
+
+    /**
+     * Returns an error type identifier for this error.
+     */
+    public String code() {
+      return code;
+    }
+
+    /**
+     * Returns the field in the request which caused the error. Might be {@code null}.
```

Comment:
Nit, Any reason for using "Might be {@code null}" instead of "an optional" like you do bellow?